grant-grant-revoke-user-role documentation (#43)
plantanapp/documentation: docs/Actions/revoke-user-role.md

---
id: revoke-user-role
title: Revoke User Role
sidebar_label: Revoke User Role
---

> Audience: [`Business Users`](/docs/audience#business-users)<br/>
> Skill Prerequisites: `Permission Scheme`

## `Typical Use Cases`

- Revoke roles of existing users
- Remove access to resources for specific users

## `Don't use it to`

> - Prevent users from accessing the system at all. Instead, mark users as deleted or unauthorize them

Reviewer: Do we have actions for unauthorize and delete to mention them here?

Author: Yes. Mentioned the two actions now.

## `Related Actions`

| Action Name                                             | Description                                                         |
| ------------------------------------------------------- | ------------------------------------------------------------------- |
| [Load Users](/docs/Actions/load-user.md)                | Can be used to revoke roles to a single user or to a list of users. |
| [User Registration](/docs/Actions/user-registration.md) | Can be used to revoke roles to a newly created user.                |

## `Input Parameter Reference`

| Parameter        | Description                                                                              | Supports Variables |
| ---------------- | ---------------------------------------------------------------------------------------- | ------------------ |

> | Role             | Select the security role, from existing roles, to revoke.                                | Yes                |

Reviewer: ```suggestion
| Role             | Select the security role to revoke from a list with all the roles in the application.                                | Yes                |
```

> | Other Role Names | Freely input the role name(s) you want to revoke. Multiple roles are separated by comma. | Yes                |

Reviewer: Can I use role ids as well?

Author: Yes, updated the description already

## `Examples`

> ### `1. Revoke a User Role `

Reviewer: It's a dangerous example. If a Citizen Developer runs this sample, they're going to lock themselves out. Let's do the example with the Users role instead.

Author: My bad, I changed the example.

> The action below loads current user using [Load User](/docs/Actions/load-user) action and revokes its '[Citizen Developers]' role. [Import it](/docs/Actions/Import-actions) into your application to see it in action.

Reviewer: ```suggestion
The action below loads the current user using [Load User](/docs/Actions/load-user) action and revokes its '[Citizen Developers]' role. [Import it](/docs/Actions/Import-actions) into your application to see it in action.
```

```json
{
    "Title": "Execute Actions",
    "ActionType": "ExecuteActions",

>     "Description": "revoke current Citizen Developers user Role",

Reviewer: ```suggestion
    "Description": "Revoke Citizen Developers role from current user",
```

                },
                "ActionType": "LoadUser",
                "$_isOpen": false,
                "$_isLoaded": true,
                "$_isFocus": true,
                "Definition": {
                    "IsClientAction": false,
                    "Settings": {
                        "Group": "Context"
                    }
                }
            },
            {
                "Id": -1,
                "$_uid": "action15918684196265931",
                "Parameters": {
                    "RoleId": {
                        "Expression": "",
                        "Value": "6",
                        "IsExpression": false,
                        "Parameters": {}
                    },
                    "RoleNames": ""
                },
                "ActionType": "RevokeUserRole",
                "$_isOpen": false,
                "$_isLoaded": true,
                "$_isFocus": true,
                "Definition": {
                    "IsClientAction": false,
                    "Settings": {
                        "Group": "User"
                    }
                }
            }
        ]
    }
}
```

### `2. Revoke a Role from all users`

> The action below loads all users using [Load Entities](/docs/Actions/load-entitites) and [Load User](/docs/Actions/load-user) action and revokes the '[Managers]' role from every one. [Import it](/docs/Actions/Import-actions) into your application to see it in action.

Reviewer: ```suggestion
The action below loads all users using [Load Entities](/docs/Actions/load-entitites) and [Load User](/docs/Actions/load-user) action and revokes the `Managers` role from every one. [Import it](/docs/Actions/Import-actions) into your application to see it in action.
```

                                "RoleId": {
                                    "Expression": "",
                                    "Value": "7",
                                    "IsExpression": false,
                                    "Parameters": {}
                                },
                                "RoleNames": ""
                            },
                            "ActionType": "RevokeUserRole",
                            "$_isOpen": false,
                            "$_isLoaded": true,
                            "$_isFocus": true,
                            "Definition": {
                                "IsClientAction": false,
                                "Settings": {
                                    "Group": "User"
                                }
                            }
                        }
                    ],
                    "ContinueOnError": "",
                    "OnError": []
                },
                "$_uid": "action15922109647731312",
                "$_isOpen": false,
                "$_isLoaded": true,
                "$_isFocus": true
            }
        ]
    }
}
```

> ## `Frequently asked questions`

Reviewer: Same as with Grant Role, remove this section and make sure the content falls naturally where it belongs.

Author: Done
